common/src/main/java/com/wynntils/commands/CompassCommand.java

```diff
@@ -0,0 +1,177 @@
+        response.append(new TextComponent(location.toString()).withStyle(ChatFormatting.WHITE));
+        context.getSource().sendSuccess(response, false);
+        return 1;
+    }
+
+    private int compassService(CommandContext<CommandSourceStack> context) {
+        String searchedName = context.getArgument("name", String.class);
+
+        List<ServiceKind> matchedKinds = Arrays.stream(ServiceKind.values())
+                .filter(kind -> StringUtils.partialMatch(kind.getName(), searchedName))
+                .toList();
+
+        if (matchedKinds.isEmpty()) {
+            MutableComponent response = new TextComponent("Found no services matching '" + searchedName + "'")
+                    .withStyle(ChatFormatting.RED);
+            context.getSource().sendFailure(response);
+            return 0;
+        }
+
+        if (matchedKinds.size() > 1) {
+            MutableComponent response = new TextComponent("Found multiple services matching '" + searchedName
+                            + "'. Pleace specify with more detail. Matching: ")
+                    .withStyle(ChatFormatting.RED);
+            response.append(new TextComponent(String.join(
+                    ", ", matchedKinds.stream().map(kind -> kind.getName()).toList())));
+            context.getSource().sendFailure(response);
+            return 0;
+        }
+
+        ServiceKind selectedKind = matchedKinds.get(0);
+
+        Vec3 currentLocation = McUtils.player().position();
+        Optional<Poi> closestServiceOptional = MapModel.getAllPois().stream()
+                .filter(poi -> poi instanceof ServicePoi servicePoi
+                        && servicePoi.getKind().equals(selectedKind))
+                .min(Comparator.comparingDouble(poi -> currentLocation.distanceToSqr(
+                        poi.getLocation().getX(),
+                        poi.getLocation().getY(),
+                        poi.getLocation().getZ())));
+        if (closestServiceOptional.isEmpty()) {
+            // This really should not happen...
+            MutableComponent response = new TextComponent("Found no services of kind '" + selectedKind.getName() + "'")
+                    .withStyle(ChatFormatting.RED);
+            context.getSource().sendFailure(response);
+            return 0;
+        }
+        Poi closestService = closestServiceOptional.get();
+        CompassModel.setCompassLocation(closestService.getLocation().asLocation());
+
+        MutableComponent response = new TextComponent("Setting compass to " + selectedKind.getName() + " at ")
+                .withStyle(ChatFormatting.AQUA);
+        response.append(new TextComponent(closestService.getLocation().toString()).withStyle(ChatFormatting.WHITE));
+        context.getSource().sendSuccess(response, false);
+        return 1;
+    }
+
+    private int compassPlace(CommandContext<CommandSourceStack> context) {
```

Comment:
This is really similar to `/territory`

Reply:
I'm planning to look how I can unite the territory code with the new MapModel. The difference here is the selection of POIs that this draws from; this here is place names from the Wynncraft web map, not territories.